exporter/modules.go

```diff
@@ -0,0 +1,52 @@
+package exporter
+
+import (
+	"strings"
+
+	"github.com/gomodule/redigo/redis"
+	"github.com/prometheus/client_golang/prometheus"
+	log "github.com/sirupsen/logrus"
+)
+
+func (e *Exporter) extractModulesMetrics(ch chan<- prometheus.Metric, c redis.Conn) {
+	info, err := redis.String(doRedisCmd(c, "INFO", "MODULES"))
+	if err != nil {
+		log.Errorf("extractSearchMetrics() err: %s", err)
+		return
+	}
+
+	lines := strings.Split(info, "\r\n")
+	for _, line := range lines {
+		log.Debugf("info: %s", line)
+
+		split := strings.Split(line, ":")
+		if len(split) != 2 {
+			continue
+		}
+
+		if split[0] == "module" {
+			// module format: 'module:name=search,ver=21005,api=1,filters=0,usedby=[],using=[ReJSON],options=[handle-io-errors]'
+			module := strings.Split(split[1], ",")
+			if len(module) != 7 {
```

Comment:
I've been thinking about this section a bit. 
I was wondering if we could make this more generic (e.g. make it work for other modules as well), if it's worth it and, if yes, how to do it.

The way this is written now, it's very much working for just the search module (the hardcoded labels, the len() == 7) but the metric is just "module"info" (I know, we talked about that) and that suggests infos for mor than just one module (and that's what could be the case in the future)

I can think of two ways forward.

1. Either make this for just search but then let's call this "module_search_info" or so, check here not only len() ==7 but that the name is "search" 

2. Make this truly generic. Generate labels on the fly by parsing, use metric names for individual modules, etc etc.

What do you think?

Reply:
Morning,
It's actually already generic and collects all modules (not only search) information that available from `info modules` output.

Reply:
I see. So there are always 7 items in the module info string? (that wasn't clear to me, I kind of expected that each module could have a different number of pieces returned)

In that case we're good here.

Reply:
No problem.

I'm not sure if it's 100% guaranteed by Redis, but so far looks like this:
```log
# Modules
module:name=RedisCompat,ver=1,api=1,filters=0,usedby=[],using=[],options=[]
module:name=redisgears_2,ver=20020,api=1,filters=0,usedby=[],using=[],options=[]
module:name=search,ver=21005,api=1,filters=0,usedby=[],using=[ReJSON],options=[handle-io-errors]
module:name=ReJSON,ver=20803,api=1,filters=0,usedby=[search],using=[],options=[handle-io-errors]
module:name=timeseries,ver=11202,api=1,filters=0,usedby=[],using=[],options=[]
module:name=bf,ver=20802,api=1,filters=0,usedby=[],using=[],options=[]
```

Reply:
I clarified comment in the code in [last commit](https://github.com/oliver006/redis_exporter/pull/953/commits/9e7822e716a6dcacf5210039fc9092a85a69d8c9)

Reply:
Thanks!